src/serialport.cpp

```diff
@@ -336,6 +336,76 @@ void EIO_AfterWrite(uv_work_t* req) {
   delete queuedWrite;
 }
 
+#ifdef WIN32
+NAN_METHOD(Read) {
+  // file descriptor
+  if (!info[0]->IsInt32()) {
+    Nan::ThrowTypeError("First argument must be an int");
+    return;
+  }
+  int fd = Nan::To<v8::Int32>(info[0]).ToLocalChecked()->Value();
+
+  // buffer
+  if (!info[1]->IsObject() || !node::Buffer::HasInstance(info[1])) {
+    Nan::ThrowTypeError("Second argument must be a buffer");
+    return;
+  }
+
+  // offset
+  if (!info[2]->IsInt32()) {
+    Nan::ThrowTypeError("Third argument must be an int");
+    return;
+  }
+  int offset = Nan::To<v8::Int32>(info[2]).ToLocalChecked()->Value();
+  
+  // bytes to read
+  if (!info[3]->IsInt32()) {
+    Nan::ThrowTypeError("Fourth argument must be an int");
+    return;
+  }
+  int bytesToRead = Nan::To<v8::Int32>(info[3]).ToLocalChecked()->Value();
+
+  // callback
+  if (!info[4]->IsFunction()) {
+    Nan::ThrowTypeError("Fifth argument must be a function");
+    return;
+  }
+
+  ReadBaton* data = new ReadBaton();
+  memset(data, 0, sizeof(ReadBaton));
+  data->fd = fd;
+  data->offset = offset;
+  data->bytesToRead = bytesToRead;
+  v8::Local<v8::Object> buffer = info[1]->ToObject();
+  data->bufferData = node::Buffer::Data(buffer);
+  data->callback.Reset(info[4].As<v8::Function>());
+
+  uv_work_t* req = new uv_work_t();
+  req->data = data;
+  uv_queue_work(uv_default_loop(), req, EIO_Read, (uv_after_work_cb)EIO_AfterRead);
+}
+
+void EIO_AfterRead(uv_work_t* req) {
+  Nan::HandleScope scope;
+  ReadBaton* data = static_cast<ReadBaton*>(req->data);
+
+  v8::Local<v8::Value> argv[2];
+    
+  if (data->errorString[0]) {
+    argv[0] = Nan::Error(data->errorString);
+    argv[1] = Nan::Undefined();
+  } else {
+    argv[0] = Nan::Null();
+    argv[1] = Nan::New<v8::Integer>((int)data->bytesRead);
```

Comment:
Ok. I had left it out since the push-methods _read wasn't using that parameter - https://github.com/EmergingTechnologyAdvisors/node-serialport/blob/master/lib/push-methods.js#L31